src/OpenLoco/src/GameCommands/CreateTown.cpp

```diff
@@ -0,0 +1,200 @@
+#include "Audio/Audio.h"
+#include "Date.h"
+#include "Economy/Expenditures.h"
+#include "GameCommands.h"
+#include "GameState.h"
+#include "Localisation/StringIds.h"
+#include "Localisation/StringManager.h"
+#include "Map/SurfaceElement.h"
+#include "Map/TileElement.h"
+#include "Map/TileManager.h"
+#include "Objects/LandObject.h"
+#include "Objects/ObjectManager.h"
+#include "S5/S5.h"
+#include "Ui/WindowManager.h"
+#include "ViewportManager.h"
+#include "World/IndustryManager.h"
+#include "World/StationManager.h"
+#include "World/TownManager.h"
+
+using namespace OpenLoco::Interop;
+using namespace OpenLoco::World;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x00496FE7
+    // TODO: only used here, but move to town manager?
+    static Town* initialiseTown(Pos2 pos)
+    {
+        registers regs;
+        regs.eax = pos.x;
+        regs.ecx = pos.y;
+        call(0x00496FE7, regs);
+
+        if (regs.esi != -1)
+            return reinterpret_cast<Town*>(regs.esi);
+        else
+            return nullptr;
+    }
+
+    // 0x004C5604
+    // TODO: used by other functions; move elsewhere
+    static uint16_t sub_4C5604(Pos2 pos)
```

Comment:
countNearbyWaterTiles

Reply:
Haha, I'd just renamed it to `countNearbyWaterTiles` as per your earlier suggestion. Saw that there was a `countSurroundingWaterTiles` already in the TileManager, indeed.